### ISSUE TITLE
cross-platform opencover for .net core on Linux/MacOS 

### ISSUE THREAD

**Issue by sawilde** [OpenCover/opencover#703]
Please provide the following information when submitting an issue, where appropriate replace the `[ ]` with a `[X]`

### My Framework

- [X] .NET Core 1.*
- [X] Something else

### My Environment

- [ ] Linux
- [ ] MacOS

### I have already...

- [X] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [ ] no coverage being recorded
- [ ] 32 or 64 bit support
- [X] feature request

### Expected Behavior

Run OpenCover on Linux or Mac OS

**Comment by sawilde** [comment]
Also wanted by the .net core team - https://github.com/dotnet/corefx/issues/17107
But also dependant on the work porting the profiling API to .net core - https://github.com/dotnet/coreclr/issues/445

**Comment by pms1969** [comment]
@sawilde How's this looking. We're just about to convert our codebase to dotnetcore2.0, and code coverage will be an important part of our build, so looking around, OpenCover looks a good choice for us.

I appreciate it might take a bit of work, and happy to do what I can to help and make this work for linux. Any pointers to get me started?

**Comment by sawilde** [comment]
@pms1969, as far as I know, the API has not yet been created/released for those platforms to allow the OpenCover work to even start.

**Comment by pms1969** [comment]
Ok, that's a shame.  I'll keep an eye out for it, and dig around the or release repos.. is it a native app, or just a .net lib?

**Comment by johnnyasantoss** [comment]
Is this already working on Linux?
I saw that the profiler is already ported to it.
[Profiler Status](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/profiling-api-status.md)

**Comment by sawilde** [comment]
@johnnyasantoss no-one is working on a linux implementation yet AFAIK - not sure when that will happen.

**Comment by FastNinja** [comment]
someone this guy managed to build a workaround - we tested and it measured code coverage on Linux for a mid size production DotNet cover project.
https://github.com/lucaslorentz/minicover
It is not a standard Microsoft way - more like a hack/workaround. but it works!

**Comment by SteveGilham** [comment]
Yet another https://github.com/SteveGilham/altcover

**Comment by giorgiozoppi** [comment]
Ok back in town to help, is there anyother willing to help on this. Just to know?

**Comment by zaiboot** [comment]
How can I help?
